packages/swapper/src/swappers/thorchain/buildThorTrade/buildThorTrade.ts

```diff
@@ -67,7 +75,7 @@ export const buildTrade = async ({
         receiveAddress: destinationAddress,
         txData: ethTradeTx.txToSign
       }
-    } else if (input.chainId === KnownChainIds.BitcoinMainnet) {
+    } else if (chainNamespace === CHAIN_NAMESPACE.Bitcoin) {
```

Comment:
Would be good for us to add type guards and/or create our union types from an array so we can narrow effectively and avoid casting types later.

ex.
```
export const evmChainIds = [KnownChainIds.EthereumMainnet, KnownChainIds.AvalancheMainnet] as const

export type EvmChainId = typeof evmChainIds[number]
```

Reply:
This should allow the check `if (utxoSupportedChainIds.includes(input.chainId)` to properly narrow the type. Can also leverage a `isUtxoSupportedChainId` typeguard to perform a similar function

Reply:
Postponing for now to think through proper pattern for narrowing types